### PR TITLE
XW-1206 | fallback nicely when title is empty

### PR DESCRIPTION
**Pull request by bognix**
@Wikia/x-wing

### CODE REVIEW

**Comment by bognix** [comment]
unit tests failure not related with the change

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 1]
```diff
@@ -224,41 +224,51 @@ public function getWikiVariables() {
...
+		$titleExists = true;
+		$dimensions = [];
 
-		if ( $article instanceof Article && $title->isRedirect() ) {
-			$title = $this->handleRedirect( $title, $article, [] )[0];
+		// Exception is thrown when empty title is send
+		// In that case we don't want to set dimensions which depend on title
+		// Title parameter is empty for URLs like /main/edit, /d etc. (all pages outside /wiki/ space)
+		try {
+			$title = $this->getTitleFromRequest();
```
Comment (vforge) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php:235]: So we're using `getTitleFromRequest` to override value from `wgTitle` (few lines above)

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 2]
```diff
@@ -224,41 +224,51 @@ public function getWikiVariables() {
...
+		} catch (Exception $ex) {
+			$titleExists = false;
+		}
+
+		if ( !empty ( $titleExists ) ) {
```
Comment (vforge) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: Can we have "smart defaults" to those `dimension`s when we don't have Title?

Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: I don't think there is a need for `$titleExists` variable. Just move what's in this block into `try { ... }` block.

File: extensions/wikia/MercuryApi/MercuryApiController.class.php (hunk 1 above)
Comment (bognix) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php:235]: resolved

**Comment by vforge** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Pull request merged**